### PR TITLE
add hasSuffix & contains & jsDocs and some refactor [std/bytes]

### PR DESCRIPTION
**Pull request by disizali**
this PR is going to :
- add  `hasSuffix` function
- add `contains` function
- add some `jsDocs` to functions
- refactor `equal & hasPrefix` functions
- new tests added

current tests were passed successfully with new refactors 
Please let me know if you know of any exceptions that may have occurred

### CODE REVIEW

File: std/bytes/mod.ts [hunk 1]
```diff
@@ -1,7 +1,11 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 import { copyBytes } from "../io/util.ts";
 
-/** Find first index of binary pattern from a. If not found, then return -1 **/
+/**
+ * Find first index of binary pattern from a. If not found, then return -1
+ * @param a soruce array
+ * @param b pattern to find in source array
```
Comment (jayphbee) [inline std/bytes/mod.ts:7]: @param pat?

File: std/bytes/mod.ts [hunk 2]
```diff
@@ -23,7 +27,11 @@ export function findIndex(a: Uint8Array, pat: Uint8Array): number {
   return -1;
 }
 
-/** Find last index of binary pattern from a. If not found, then return -1 **/
+/**
+ * Find last index of binary pattern from a. If not found, then return -1.
+ * @param a soruce array
+ * @param b pattern to find in source array
```
Comment (jayphbee) [inline std/bytes/mod.ts:33]: pat ?

File: std/bytes/mod.ts [hunk 3]
```diff
@@ -45,21 +53,31 @@ export function findLastIndex(a: Uint8Array, pat: Uint8Array): number {
   return -1;
 }
 
-/** Check whether binary arrays are equal to each other **/
+/**
+ * Check whether binary arrays are equal to each other.
+ * @param a First array to check
+ * @param b Second array to check
```
Comment (jayphbee) [inline std/bytes/mod.ts:59]: match ?

File: std/bytes/mod.ts [hunk 4]
```diff
@@ -45,21 +53,31 @@ export function findLastIndex(a: Uint8Array, pat: Uint8Array): number {
...
-  }
-  return true;
+  return String(a) === String(match);
 }
 
-/** Check whether binary array has binary prefix  **/
+/**
+ * Check whether binary array starts with prefix.
+ * @param a First array to concatenate
+ * @param b Second array to concatenate
```
Comment (jayphbee) [inline std/bytes/mod.ts:68]: prefix ?

File: std/bytes/mod.ts (hunk 1 above)
Comment (disizali) [reply std/bytes/mod.ts]: good calls @jayphbee, thanks 👍

File: std/bytes/mod.ts [hunk 5]
```diff
@@ -45,36 +53,52 @@ export function findLastIndex(a: Uint8Array, pat: Uint8Array): number {
...
-    if (a[i] !== match[i]) return false;
-  }
-  return true;
+/**
+ * Check whether binary arrays are equal to each other.
+ * @param source first array to check equality
+ * @param match second array to check equality
+ */
+export function equal(source: Uint8Array, match: Uint8Array): boolean {
+  return String(source) === String(match);
```
Comment (kitsonk) [inline std/bytes/mod.ts:62]: Comparing by coercing to strings is slower in every imaginable way.  See: https://jsperf.com/uint8compare/1

Why are you attempting to change this?

Comment (disizali) [reply std/bytes/mod.ts]: I think it was cleaner, didn't check for performance actually.
reverted
thanks for notice that 👍

File: std/bytes/mod.ts [hunk 6]
```diff
@@ -45,36 +51,56 @@ export function findLastIndex(a: Uint8Array, pat: Uint8Array): number {
...
+    if (source[i] !== prefix[i]) return false;
   }
   return true;
 }
 
-/**
- * Repeat bytes. returns a new byte slice consisting of `count` copies of `b`.
- * @param b The origin bytes
+/** Check whether binary array ends with suffix.
+ * @param source srouce array
```
Comment (cmorten) [inline std/bytes/mod.ts:78]: `srouce` -> `source`

**Pull request merged**